go/libraries/doltcore/doltdb/errors.go

```diff
@@ -82,6 +82,9 @@ func IsNotFoundErr(err error) bool {
 }
 
 func IsNotACommit(err error) bool {
+	if errors.Is(err, ErrBranchNotFound) {
```

Comment:
If this is necessary for this type, do it for the other types in the case as well, otherwise it will break when we begin wrapping those too